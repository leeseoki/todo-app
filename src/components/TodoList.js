import React, {Component} from 'react'
import TodoItem from './TodoItem'

export default class TodoList extends Component {
  render () {
    var items = this.props.items.map((item, index) => {
      return (
        <TodoItem key={index} item={item} id={index} removeItem={this.props.removeItem} maskItemDone={this.props.maskItemDone} />
      );
    });
    return (
      <ul className="list-group"> 
        <li className="list-group-item active">My Todo</li>
        {items} 
      </ul>
    );
  }
}
