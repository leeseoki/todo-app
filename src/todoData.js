let todoData = [];
todoData.push({id: 1, title: "eat", done: false});
todoData.push({id: 2, title: "sleep", done: true});
todoData.push({id: 3, title: "code", done: true});
export default todoData
