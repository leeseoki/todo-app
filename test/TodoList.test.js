import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import TodoList from '../src/components/TodoList'
import todoData from '../src/todoData'

describe('TodoList Component', () => {

  beforeAll(() => {
      Enzyme.configure({ adapter: new Adapter() })
  })

  it('renders correctly', () => {
      const component = shallow(<TodoList items={todoData} />)
      expect(component).toMatchSnapshot()
  })
})

