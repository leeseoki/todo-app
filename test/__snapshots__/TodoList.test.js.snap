// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    items={
      Array [
        Object {
          "done": false,
          "id": 1,
          "title": "eat",
        },
        Object {
          "done": true,
          "id": 2,
          "title": "sleep",
        },
        Object {
          "done": true,
          "id": 3,
          "title": "code",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="list-group-item active"
        >
          My Todo
        </li>,
        Array [
          <TodoItem
            id={0}
            item={
              Object {
                "done": false,
                "id": 1,
                "title": "eat",
              }
            }
          />,
          <TodoItem
            id={1}
            item={
              Object {
                "done": true,
                "id": 2,
                "title": "sleep",
              }
            }
          />,
          <TodoItem
            id={2}
            item={
              Object {
                "done": true,
                "id": 3,
                "title": "code",
              }
            }
          />,
        ],
      ],
      "className": "list-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Todo",
          "className": "list-group-item active",
        },
        "ref": null,
        "rendered": "My Todo",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "id": 0,
          "item": Object {
            "done": false,
            "id": 1,
            "title": "eat",
          },
          "maskItemDone": undefined,
          "removeItem": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "id": 1,
          "item": Object {
            "done": true,
            "id": 2,
            "title": "sleep",
          },
          "maskItemDone": undefined,
          "removeItem": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "id": 2,
          "item": Object {
            "done": true,
            "id": 3,
            "title": "code",
          },
          "maskItemDone": undefined,
          "removeItem": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="list-group-item active"
          >
            My Todo
          </li>,
          Array [
            <TodoItem
              id={0}
              item={
                Object {
                  "done": false,
                  "id": 1,
                  "title": "eat",
                }
              }
            />,
            <TodoItem
              id={1}
              item={
                Object {
                  "done": true,
                  "id": 2,
                  "title": "sleep",
                }
              }
            />,
            <TodoItem
              id={2}
              item={
                Object {
                  "done": true,
                  "id": 3,
                  "title": "code",
                }
              }
            />,
          ],
        ],
        "className": "list-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Todo",
            "className": "list-group-item active",
          },
          "ref": null,
          "rendered": "My Todo",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "id": 0,
            "item": Object {
              "done": false,
              "id": 1,
              "title": "eat",
            },
            "maskItemDone": undefined,
            "removeItem": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "id": 1,
            "item": Object {
              "done": true,
              "id": 2,
              "title": "sleep",
            },
            "maskItemDone": undefined,
            "removeItem": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "id": 2,
            "item": Object {
              "done": true,
              "id": 3,
              "title": "code",
            },
            "maskItemDone": undefined,
            "removeItem": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
